src/components/VmDetails/cards/DisksCard/DiskImageEditor.js

```diff
@@ -0,0 +1,428 @@
+
+    return <React.Fragment>
+      {React.cloneElement(trigger, { onClick: this.open })}
+
+      <Modal
+        dialogClassName={style['editor-modal']}
+        id={`${idPrefix}-modal`}
+        show={this.state.showModal}
+        onHide={this.close}
+      >
+        <Modal.Header>
+          <Modal.CloseButton id={`${idPrefix}-modal-close`} onClick={this.close} />
+          <Modal.Title>{createMode ? msg.createNewDisk() : msg.editDisk()}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+
+          <Form horizontal>
+            {/* Alias */}
+            <FormGroup controlId={`${idPrefix}-alias`} className={createMode && 'required'}>
+              <LabelCol sm={3}>
+                { msg.diskEditorAliasLabel() }
+              </LabelCol>
+              <Col sm={9}>
+                <FormControl
+                  type='text'
+                  defaultValue={this.state.alias}
+                  onChange={this.changeAlias}
+                />
+              </Col>
+            </FormGroup>
+
+            {/* Size Display (for edit mode) */}
+            { !createMode &&
+            <FormGroup controlId={`${idPrefix}-size-info`}>
+              <LabelCol sm={3}>
+                { msg.diskEditorSizeLabel() }
+                { !isImage &&
+                  <FieldLevelHelp
+                    inline
+                    content={msg.diskEditorSizeCantChangeHelp()}
+                    buttonClass={style['editor-field-help']}
+                  />
+                }
+              </LabelCol>
+              <Col sm={9}>
+                <div id={`${idPrefix}-size-info`} className={style['editor-field-read-only']}>
+                  { diskSize / 1024 ** 3 }
+                </div>
+              </Col>
+            </FormGroup>
+            }
+
+            {/* Size Editor (initial size for create, expand by size for edit) */}
+            { (createMode || isImage) &&
+            <FormGroup controlId={`${idPrefix}-size`} className={createMode && 'required'}>
+              <LabelCol sm={3}>
+                { createMode &&
+                  <React.Fragment>
+                    {msg.diskEditorSizeLabel()}
+                    <FieldLevelHelp
+                      inline
+                      content={msg.diskEditorSizeCreateHelp()}
+                      buttonClass={style['editor-field-help']}
+                    />
+                  </React.Fragment>
+                }
+                { !createMode && msg.diskEditorResizeLabel() }
+              </LabelCol>
+              <Col sm={9}>
+                { (createMode || isImage) &&
+                  <FormControl
+                    type='number'
+                    value={this.state.size}
+                    onChange={this.changeSize}
+                  />
+                }
+                { !createMode && !isImage &&
+                  <div id={`${idPrefix}-size-readonly`} className={style['editor-field-read-only']}>
```

Comment:
For consistency, please, use `${idPrefix}-size` for readonly values, and `${idPrefix}-size-edit` for form fields.